### PR TITLE
J978li fix conc state and basic state

### PR DESCRIPTION
**Pull request by Simonliuwaterloo**
* For basic states, leave _state to `true`
* For states that have conc children, use `_states` which is a list of states

### CODE REVIEW

File: org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm [hunk 1]
```diff
@@ -26,7 +27,8 @@ class State(ABC):
         # State() => doesn't do anything
         # State(parent_state: State) => set _parent_state
         # State(parent_state: State, root_state: State) => also set _root_state
-        self._state = None
+        self._state: State = None # used when there is only one child state
```
Comment (AidanCampbell738) [inline org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm]: I would reword the comment to "used when there is only one active child state". Otherwise it sounds like _state is only used for states with exactly one child state (when it can be used for states with multiple children like Bit1 from Bitcounter).

File: org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm [hunk 2]
```diff
@@ -26,7 +27,8 @@ class State(ABC):
...
+        self._states: List[State] = [] # used when there is multple children conc states
```
Comment (AidanCampbell738) [inline org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm]: It might be valuable to rename _state and _states to something more descriptive like _active_state and _conc_states respectively.

**Review by AidanCampbell738 — APPROVED** [review]

**Review by Chestshire — APPROVED** [review]
LGTM

**Review by RuomeiYan — APPROVED** [review]
